Makefile

```diff
@@ -0,0 +1,11 @@
+# It seems that the venv doesn't work in the makefile. https://stackoverflow.com/questions/44052093/makefile-with-source-get-error-no-such-file-or-directory
+# If you want to use virtual environment, you can create the virtual environment first, then run the make file, here is the example
```

Comment:
@kevinyu98 -- this `Makefile` will create the Python3 virtual environment if it does not exist and install all requirements, and re-install additional dependencies if `setup.py` was modified:

```Make
# Copyright 2020 kubeflow.org
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

VENV = .venv
export VIRTUAL_ENV := $(abspath ${VENV})
export PATH := ${VIRTUAL_ENV}/bin:${PATH}

# create virtual environment if it does not exist, (re-)install requirements if setup.py was changed
venv: $(VENV)/bin/activate
$(VENV)/bin/activate: sdk/python/setup.py
	test -d $(VENV) || python3 -m venv $(VENV)
	test -e $(VENV)/bin/dsl-compile-tekton || pip install -e sdk/python
	touch $(VENV)/bin/activate


test: venv
	sdk/python/tests/run_tests.sh

report:
	sdk/python/tests/test_kfp_samples.sh

```

You can see the `test` target depends on `venv`. This will ensure tests are run inside the virtual environment and if it does not exist, creates it and installs `kfp` and `kfp-tekton` with its dependencies.

Reply:
@ckadner thanks, I tried, but it seems not working in my environment. After the run, my command line still shows it is in the `base`, not `venv`. Seems that we have to run the activate on the shell command line, something like this made it work. I will submit the code for reviewing.
`. $(VENV)/activate && exec $(notdir $(SHELL))`

Reply:
@kevinyu98 -- the "virtual environment" will only be used for the sub-shell of the Make target. The console shell from which you start `make` will not be affected by design. I have not used the `source bin/activate` method in the example. Instead I exported the environment variables `VIRTUAL_ENV` and `PATH` for all sub-shells of the make process.

Reply:
@ckadner I see. I was trying to find a way to activate the `venv`, then the `venv` will stay there until `deactivate`, just like what we execute those commands without the makefile on the console. I haven't explored using the environment variable for all sub-shells ways. I just submit the revised code, can you help review? thanks